### ISSUE TITLE
QUIC: Detecting whether QUIC is supported

### ISSUE THREAD

**Issue by JamesNK** [dotnet/runtime#52145]
QUIC on Windows (and perhaps other operating systems) will have OS and other dependency requirements.

It would be useful if there is an API to check whether it is supported. For example:

```cs
if (QuicConnection.IsSupported)
{
    // Do QUIC things with System.Net.Quic
}
else
{
    // Fallback to doing TCP things with Socket
}
```

It would return false in Blazor WebAssembly. It would return false in Windows if the OS doesn't have schannel support. Etc.

**Comment by geoffkizer** [comment]
There is some code in SocketsHttpHandler that does this sort of test currently: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L867

Note the comment referencing https://github.com/dotnet/runtime/issues/44922; I think that is the eventual solution here, we are just waiting for this to land.

**Comment by ManickaP** [comment]
Windows version check fixed with #54488

Note that S.N.Quic is PNSE on anything non Win/Linux/OSX, e.g. wasm. Thanks to code analyzers you get compilation errors when trying to reference and use S.N.Quic on those platforms.
